### ISSUE TITLE
plz cover for python3 seg fault 11

### ISSUE THREAD

**Issue by bdangit** [thought-machine/please#1700]
I continously get seg fault errors when running `plz cover`, specifically for a python3 component.

## Logs

```
$ plz --version
Please version 15.17.1

$ plz cover //{redacted}:tests
Fail: //{redacted}:tests   0 passed   0 skipped   0 failed   1 errored Took 270ms
Error: TestFailed in tests
Test failed
signal: segmentation fault
//{redacted}:tests 1 test run in 271ms; 0 passed, 1 errored
    tests  ERROR
1 test target and 1 test run; 0 passed, 1 errored.
Total time: 320ms real, 270ms compute.
Messages:
15:58:08.703   ERROR: //{redacted}:tests failed: Test failed
Coverage results:
{redacted}:
  {redacted}.py 0/29 lines, 0.0%
Total coverage:  0/29 lines, 0.0%
```

## Things I tried

I ran with `--shell` just to see if I am able to run the `tests.pex`.  That results in segfault from inside the bash shell.

```
$ plz cover //{redacted}:tests --shell
Temp directories prepared, total time 60ms:
  //{redacted}:tests: /{redacted}/plz-out/tmp/{redacted}/tests._test/run_1
    Command: $TEST

bash-5.1$ ./
components/ tests.pex
bash-5.1$ ./tests.pex
Segmentation fault: 11
```

I ran the `tests.pex` outside plz bash shell.
```
$ cd plz-out/bin/{redacted}
$ rm -rf test.results
$ ./tests.pex
✗ ./tests.pex
========================================================================= test session starts =========================================================================
platform darwin -- Python 3.9.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /{redacted}
plugins: pyfakefs-4.3.3
collected 0 items

========================================================================== warnings summary ===========================================================================
tests.pex/.bootstrap/_pytest/junitxml.py:436
  /{redacted}/tests.pex/.bootstrap/_pytest/junitxml.py:436: PytestDeprecationWarning: The 'junit_family' default value will change to 'xunit2' in pytest 6.0.
  Add 'junit_family=xunit1' to your pytest.ini file to keep the current format in future versions of pytest and silence this warning.

-- Docs: https://docs.pytest.org/en/latest/warnings.html
------------------------ generated xml file: /{redacted}/test.results/results.xml -------------------------
========================================================================= 1 warning in 0.02s ==========================================================================
ERROR: file not found: {redacted}.py
```

## Expectation

running `plz cover` should not seg fault.

**Comment by Tatskaari** [comment]
Does this happen for all python tests or just that one? Is there any C bindings going on? What third party libraries go into that test?

**Comment by bendang-okta** [comment]
I haven't been successful with running any `plz cover` for any python3 code I've written.  As far as any py deps etc, I created a dep-less test.   I still run into seg faults.

https://github.com/bdangit/plz-cover-py3

Machine wise I'm running on:

MacOS Catalina - 10.15.7
Python 3.9.2

**Comment by bdangit** [comment]
I haven't been successful with running any `plz cover` for any python3 code I've written.  As far as any py deps etc, I created a dep-less test.   I still run into seg faults.

https://github.com/bdangit/plz-cover-py3

Machine wise I'm running on:

MacOS Catalina - 10.15.7
Python 3.9.2

**Comment by Tatskaari** [comment]
Hey, sorry for the slow reply. This wasn't an easy one to debug, and i needed to get access to a mac. It looks like the issue is with our coverage wheel. We embed .so files for various versions of python into a single wheel however the python 3.4 version embeds a non-versioned .so file which was being picked up erroneously for 3.9 resulting in a segfault. I've opened a PR for visibility however I probably need to build a newer wheel.
